packages/starknet-snap/src/ui/components/ExecuteTxnUI.tsx

```diff
@@ -0,0 +1,207 @@
+    ) {
+      const amount = BigInt(call.transferAmount); // Convert to BigInt
+      if (!tokenTotals[call.transferTokenSymbol]) {
+        tokenTotals[call.transferTokenSymbol] = {
+          amount: BigInt(0),
+          decimals: call.transferTokenDecimals,
+        };
+      }
+      tokenTotals[call.transferTokenSymbol].amount += amount;
+    }
+  });
+
+  // Add the fee to the corresponding token
+  const feeTokenAmount = BigInt(maxFee);
+  if (tokenTotals[selectedFeeToken]) {
+    tokenTotals[selectedFeeToken].amount += feeTokenAmount;
+  } else {
+    // We derive decimals based on the fee token. Currently, both supported fee tokens, ETH and STRK, use the standard 18 decimals.
+    // Therefore, we use DEFAULT_DECIMAL_PLACES set to 18 here. If additional fee tokens with different decimals are introduced,
+    // this logic should be updated to handle token-specific decimals dynamically.
+    tokenTotals[selectedFeeToken] = {
+      amount: feeTokenAmount,
+      decimals: DEFAULT_DECIMAL_PLACES,
+    };
+  }
+
+  return tokenTotals;
+};
+
+/**
+ * A component for executing transactions, providing details and options to configure the transaction.
+ *
+ * @param props - The component props.
+ * @param props.signer - The signer for the transaction.
+ * @param props.chainId - The ID of the chain for the transaction.
+ * @param props.maxFee - The maximum fee allowed for the transaction.
+ * @param props.calls - The calls involved in the transaction.
+ * @param props.selectedFeeToken - The token used for fees.
+ * @param props.includeDeploy - Whether to include account deployment in the transaction.
+ * //param props.errors : TODO : add this param
+ * @returns The ExecuteTxnUI component.
+ */
+export const ExecuteTxnUI: SnapComponent<ExecuteTxnUIProps> = ({
+  signer,
+  chainId,
+  maxFee,
+  calls,
+  selectedFeeToken,
+  includeDeploy,
+  // errors, // TODO: include this later
+}) => {
+  // Calculate the totals using the helper
+  const tokenTotals = calculateTotals(calls, maxFee, selectedFeeToken);
+
+  return (
+    <Container>
+      <Box>
+        <Section>
+          <AddressUI label="Signer" address={signer} chainId={chainId} />
+        </Section>
+
+        {/* Loop through each call and render based on isTransfer */}
+
+        {calls.map((call) => (
+          <Section>
+            {call.isTransfer ? (
+              <AddressUI
+                label="Token Transfer"
+                address={call.contractAddress}
+                chainId={chainId}
+              />
+            ) : (
+              <AddressUI
+                label="Contract Interaction"
+                address={call.contractAddress}
+                chainId={chainId}
+              />
+            )}
+            {call.transferAmount &&
+            call.transferSenderAddress &&
+            call.transferRecipientAddress &&
+            call.isTransfer ? (
+              <Section>
+                {call.transferSenderAddress === signer ? null : (
```

Comment:
i think we should display the sender address anyway? wdyt

Reply:
In a transfer call the sender Address is the second element of the call data. Not sure this is generalisable to any contract interaction.

Reply:
the question is why we skip the sender? i think we should always display the sender if the contract is a transfer ?

Reply:
signer is shown anyway. Let's remove this part as its not possible to sign a transaction who's sender is not us ^^